CNV-80339: DataVolume docs (#117496)
openshift/openshift-docs: modules/con-about-specifying-cpu-architecture-datavolume-imports.adoc

// Module included in the following assemblies:
//
// * virt/creating_vm/virt-import-dv-vms.adoc

:_mod-docs-content-type: CONCEPT
[id="con-about-specifying-cpu-architecture-datavolume-imports_{context}"]
= CPU architecture selection for `DataVolume` imports

[role="_abstract"]
You can target a specific CPU architecture when you import a virtual machine (VM) image from an Open Container Initiative (OCI) image index by setting the `platform.architecture` field in the `DataVolume` custom resource (CR).

> :FeatureName: Specifying CPU architecture for DataVolume imports

Author: ```suggestion
:FeatureName: Specifying CPU architecture for data volume imports
```

> You can target a specific CPU architecture when you import a virtual machine (VM) image from an OCI image index by setting the `platform.architecture` field in the `DataVolume` spec.

Author: ```suggestion
You can target a specific CPU architecture when you import a virtual machine (VM) image from an OCI image index by setting the `platform.architecture` field in the `DataVolume` custom resource (CR).
```

Author: Done.

> You can target a specific CPU architecture when you import a virtual machine (VM) image from an OCI image index by setting the `platform.architecture` field in the `DataVolume` custom resource (CR).

Reviewer: The short description must precede any admonitions. The Technology Preview `[IMPORTANT]` admonition (injected by the `snippets/technology-preview.adoc` include on line 10) currently appears before the `[role="_abstract"]` paragraph, which violates the [RH SSG short description guidance](https://redhat-documentation.github.io/supplementary-style-guide/#shortdesc). This suggestion also expands "OCI" on first use in this module, as required by the IBM Style Guide.
```suggestion

[role="_abstract"]
You can target a specific CPU architecture when you import a virtual machine (VM) image from an Open Container Initiative (OCI) image index by setting the `platform.architecture` field in the `DataVolume` custom resource (CR).

:FeatureName: Specifying CPU architecture for DataVolume imports
include::snippets/technology-preview.adoc[]
```

In heterogeneous clusters where nodes run different CPU architectures, importing the correct image variant is essential.
Without specifying the target architecture, the imported image might not match the architecture of the node where the VM is scheduled to run.
The `platform.architecture` field resolves this by selecting the appropriate variant from the image index.

The following CPU architectures are supported:

* `amd64`
* `arm64`
* `s390x`

When you set the `platform.architecture` field in combination with `pullMethod: node`, {VirtProductName} adds a node selector to the importer pod.
This node selector ensures that the importer pod runs on a node with the matching CPU architecture.
The `node` pull method uses the container runtime cache of the node where the importer pod runs, so the pod must run on a node with the correct architecture.
Without the node selector, the importer pod might run on a node with a different architecture, resulting in a mismatch between the pulled image and the target platform.

> .Example `DataVolume` manifest with `platform.architecture`

Author: ```suggestion
Example `DataVolume` CR:
```

Author: Done.
